staging/src/github.com/kubeedge/beehive/pkg/core/model/message.go

```diff
@@ -169,12 +175,53 @@ func (msg *Message) BuildHeader(ID, parentID string, timestamp int64) *Message {
 	return msg
 }
 
-//FillBody fills message  content that you want to send
-func (msg *Message) FillBody(content interface{}) *Message {
+// SetContent sets the message content
+func (msg *Message) SetContent(content Content) *Message {
 	msg.Content = content
 	return msg
 }
 
+//FillBody fills message content that you want to send
+func (msg *Message) FillBody(val interface{}) *Message {
+	if val == nil {
+		msg.SetContent(nil)
+	} else {
+		msg.SetContent(NewContent(val))
+	}
+	return msg
+}
+
+func (msg Message) MarshalJSON() ([]byte, error) {
```

Comment:
`func (msg Message) MarshalJSON() ([]byte, error) {` is `msg` read-only

Reply:
Temporary struct is used to correspond to `UnmarshalJSON `, otherwise I am afraid that others will change Struct, but the temporary struct in `UnmarshalJSON` is not changed, which causes the codec to be different.